p2panda-sync/src/log_sync.rs

```diff
@@ -319,25 +347,24 @@ mod tests {
     }
 
     #[derive(Clone, Debug)]
-    struct LogIdTopicMap(HashMap<TopicId, Vec<String>>);
+    struct LogHeightTopicMap(HashMap<String, Logs<u64>>);
 
-    impl LogIdTopicMap {
+    impl LogHeightTopicMap {
         pub fn new() -> Self {
-            Self(HashMap::new())
+            LogHeightTopicMap(HashMap::new())
         }
 
-        fn insert(&mut self, topic: TopicId, log_ids: Vec<String>) -> Option<Vec<String>> {
-            self.0.insert(topic, log_ids)
+        fn insert(&mut self, topic: &str, logs: Logs<u64>) -> Option<Logs<u64>> {
+            self.0.insert(topic.to_string(), logs)
```

Comment:
Rather than encoding every struct as a `String` (using `Display`) we could make `Topic: StdHash` and let the (fast) hashing algorithm do the work